### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by balanalina**

### CODE REVIEW

File: app/controllers/admin/products_controller.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  private
+
+  def product_params
+    params.require(:product).permit(:title, :category, :price, :image)
+  end
+
+
+  def set_product
+    @product = Product.find(params[:id])
+  end
```
Comment (VictorMotogna) [inline app/controllers/admin/products_controller.rb:54]: we usually add another tab for private methods:
```ruby
class
  def

  private

    def
```

File: app/controllers/order_items_controller.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class OrderItemsController < ApplicationController
+  before_action :set_order_item, only: :show
+  def show ;end
```
Comment (VictorMotogna) [inline app/controllers/order_items_controller.rb:3]: you should add another new line between defining `before_action` and methods

File: app/controllers/orders_controller.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
+class OrdersController < ApplicationController
+  before_action :order_params, only: %i[create]
+  before_action :set_current_cart, only: :create
+  def new
```
Comment (VictorMotogna) [inline app/controllers/orders_controller.rb:4]: same here with the new line. improves readability a lot

File: app/controllers/password_resets_controller.rb [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    elsif params[:user][:password].length < 7
+      @user.errors.add(:password, 'Password too short')
+      render 'edit'
+    elsif @user.update(user_params)
+      reset_session
+      log_in @user
+      @user.update_attribute(:reset_digest, nil)
+      flash[:success] = 'Password has been reset!'
+      redirect_to @user
+    end
```
Comment (VictorMotogna) [inline app/controllers/password_resets_controller.rb:36]: these errors should be part of the model class.

inside `user.rb` you can have:


```ruby
validate :validate_password

private

  def validate_password
    errors.add('too short') if password.length < 7
    errors.add("can't be empty") if password..empty?
  end
```

You can also add custom validations only on update if you'd have anything special (this is just an example):
```ruby
validates :password, presence: true, on: :update, if: :password_digest_changed?
```

File: app/models/order.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  before_create :create_qr_token
+  belongs_to :user
+  has_many :order_items
+
+  def total_price
+    order_items.to_a.sum { |item| item.total_price }
+  end
+
+  def create_qr_token
+    self.qr_token = User.new_token
```
Comment (florinionce) [inline app/models/order.rb:11]: why do you use `User.new_token` here? You could use rails's builtin `has_secure_token`

File: app/views/admin/users/index.html.haml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+-provide(:title, "All users")
+%h1.edit-title All Users
+%ul.row
+  -@users.each do |user|
+    = render user
+  = will_paginate @users
+
```
Comment (florinionce) [inline app/views/admin/users/index.html.haml:7]: extra space 🤣

File: app/views/carts/show.haml [hunk 7]
```diff
@@ -1,13 +1,14 @@
...
+    = render(@cart.cart_items)
     .column
-      =button_to 'Empty Cart', @cart, method: :delete, data: { confirm: "Are you sure?"}, class: "button is-danger empty-cart"
+      = button_to 'Empty Cart', @cart, method: :delete, data: { confirm: "Are you sure?"}, class: "button is-danger empty-cart"
+      = link_to 'Place Order', new_order_url, class: "button is-danger empty-cart"
       .column.total.has_text_right
         %h4.title.is_4
           %span.f5 Total:
-          =number_to_currency(@cart.total_price)
+          = number_to_currency(@cart.total_price)
```
Comment (florinionce) [inline app/views/carts/show.haml:14]: 👏

File: app/views/orders/index.haml [hunk 8]
```diff
@@ -0,0 +1,6 @@
+-provide(:title, "My orders")
+%h1.edit-title My Orders
+%ul.row
+  -@orders.each do |order|
```
Comment (florinionce) [inline app/views/orders/index.haml:4]: 👀

File: app/views/orders/show.html.haml [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+%h3.edit-title= @order.address
+%h3.edit-title= @order.phone_number
+%h3.edit-title= @order.created_at.to_date
+%h3.edit-title= @order.created_at.to_time.strftime("%I:%M%p")
+= render(@order.order_items)
+.column.total.has_text_right
+  %h4.title.is_4
+    %span.f5 Total:
+    = number_to_currency(@order.total_price)
+
```
Comment (florinionce) [inline app/views/orders/show.html.haml:16]: 👀

File: db/seeds.rb [hunk 10]
```diff
@@ -21,8 +21,13 @@
...
   image = hash[:image]
   file_path = Rails.root.join(path.first, path.second, path.third, image)
-  new_product = Product.create_or_find_by!(title: title, category: category, price: price)
+  new_product = Product.new(title: title, category: category, price: price)
   new_product.image.attach(io: File.open(file_path), filename: image)
+  new_product.save
 end
 
+User.create(name: 'buna siua', email: 'alina@jmek.com', password: '1111111', activated: true, admin: true)
+
```
Comment (florinionce) [inline db/seeds.rb:30]: you have an extra space at the end of the file 🤣

**Review by florinionce — COMMENTED** [review]
Nice 👍

**Pull request merged**